### DISCUSSION TITLE
Individual mods for different users

### DISCUSSION THREAD

**Discussion by mad0x20wizard** [ramensoftware/windhawk#320] [Ideas]
I just stumbled upon WindHawk and find it very refreshing to change the apearance (among other things) of boring old windows programs. But something I really don't understand, either because it isn't possible or just not a use case for many people.

Personally I use Windows as a multi-user system (at least I think that's what it was designed for), using things like RDP Wrapper (another cool windows tweak) to even run multiple graphical user sessions in parallel. Even without that, I could imagine multiple people sharing one physical system either one after the other, or parallel with fast user switching. But as far as I can tell, Windhawk does not distinguish between different users (or user sessions), either in global or portable mode. The latter truly does not write into any global areas, like the registry, but as soon as it starts, every process with matching image name gets modded, regardlessly of it's session id or user name.

Is there something I am missing here? The possibility to customize windows to ones personal taste becomes a somewhat forced thing, if every user "benefits" from the same set of mods, and cannot express his or her own needs. Has it something to do with the way the mods get injected into the processes? When I open the "loaded mods" window from the tray icon, I can see there every individual process (from different sessions) with a status (usually "loaded"). From that observation I would guess it should be possible to include a specific set of mods only for processes of a individual session/user.

![image](https://github.com/user-attachments/assets/4cd85c29-cb38-47b1-a8c3-34ef3bac0ca1)

Please tell me that I am just too tired to see the switch to do that. 🥱

**Comment by m417z** [comment]
Currently there's no way to customize mod settings per-user. It's on my list to implement it in one of the future versions. Thanks for sharing your feedback and emphasizing the importance of this ability.

> But as far as I can tell, Windhawk does not distinguish between different users (or user sessions), either in global or portable mode. The latter truly does not write into any global areas, like the registry, but as soon as it starts, every process with matching image name gets modded, regardlessly of it's session id or user name.

While not a perfect solution, running Windhawk portable in non-elevated mode should affect only the processes of the current user, since it normally doesn't have permissions to inject code into processes of other users. In some environments, all processes are running elevated and UAC is not enabled, in which case this won't help.

**Comment by imagicalFantasy** [comment]
I also want per users mods config way to use Winhawk. 
Currently I made some hack using portable version but result have some issues.

So real multi users support could be very usefull.
